rules/errors.js

```diff
@@ -5,77 +5,169 @@ module.exports = {
     'no-empty': 2,
 
+    // When an error is caught using a `catch` block, it's possible to
+    // accidentally (or purposely) overwrite the reference to the error. This
+    // makes it impossible to track the error from that point on.
     // http://eslint.org/docs/rules/no-ex-assign
     'no-ex-assign': 2,
 
+    // No point in having unnecessary typecasts.
     // http://eslint.org/docs/rules/no-extra-boolean-cast
     'no-extra-boolean-cast': 0,
 
+    // No point in cluttering up the code with pointless semi-colons.
     // http://eslint.org/docs/rules/no-extra-semi
     'no-extra-semi': 2,
 
+    // Overwriting/reassigning a function written as a FunctionDeclaration is
+    // often indicative of a mistake or issue.
     // http://eslint.org/docs/rules/no-func-assign
     'no-func-assign': 2,
 
+    // In JavaScript, prior to ES6, a function declaration is only allowed in
+    // the first level of a program or the body of another function, though
+    // parsers sometimes erroneously accept them elsewhere. This only applies to
+    // function declarations; named or anonymous function expressions can occur
+    // anywhere an expression is permitted.
     // http://eslint.org/docs/rules/no-inner-declarations
     'no-inner-declarations': 2,
 
+    // Validates string arguments passed to the RegExp constructor. Seems like
+    // a sensible, extra safeguard to have.
     // http://eslint.org/docs/rules/no-invalid-regexp
     'no-invalid-regexp': 2,
 
+    // Invalid or irregular whitespace causes issues with ECMAScript 5 parsers
+    // and also makes code harder to debug in a similar nature to mixed tabs and
+    // spaces. Invalid whitespace is any whitespace character that is not a
+    // normal tab or space.
     // http://eslint.org/docs/rules/no-irregular-whitespace
     'no-irregular-whitespace': 2,
 
+    // ECMAScript provides several global objects that are intended to be used
+    // as-is. Some of these objects look as if they could be constructors due
+    // their capitalization (such as Math and JSON) but will throw an error if
+    // you try to execute them as functions. So catch these errors early on.
     // http://eslint.org/docs/rules/no-obj-calls
     'no-obj-calls': 2,
 
+    // Properties that are not valid identifiers need to be quoted. For example
+    // the property `foo-bar` needs quotes, while `fooBar` does not. Generally
+    // quotes are extra and unecessary since the majority of properties should
+    // be camel-cased. However, sometimes you need those odd-named properties
+    // and, in those cases, force people to quote all their properties. This
+    // intentionally causes a bit more of a headache, trying to push people in
+    // the direction of property naming their properties and highlighting the
+    // non-conformant properties as generally part of a different domain.
     // http://eslint.org/docs/rules/quote-props
     'quote-props': [ 2, 'consistent-as-needed' ],
```

Comment:
I prefer `'as-needed'` - that way you don't have to change the quoting of an entire object if you add/remove an invalid identifier.